### DISCUSSION TITLE
Next/Prev Navigation

### DISCUSSION THREAD

**Discussion by jamesrice96** [algolia/instantsearch#6453] [Q&A]
I have a list/detail view for my search results and I want to add a next/prev navigation on the details page.

The next/prev is based on the refinements made on the previous screen. 

For example, I search for "Sony TVs" and 3 items are returned.

I click through to the 2nd item and can navigate to the ones "either side" of the current item.

Does anyone know if there's a straight forward way of doing this?

I'm currently using the `useInstantSearch()` hook to get the filtered results. 

But this is limited to the items on a "page" meaning if there are a lot of results we don't get a full list to cycle through.

I can up the limit but I'm using this with typesense and they have a hard limit of 250 items.

There's a chance the filters could exceed that 🤔 

There's information about the page numbers returned on the hook but just wondering if there's an easier method to achieving this.

Thanks

**Comment by Haroenv** [comment]
Are you looking for the usePagination hook, so you can go to the next and previous page with that hook? not sure what your problem is without reproduction

**Reply by jamesrice96** [reply]
Thanks for the reply @Haroenv 

Yeah, I initially thought I needed the `usePagingation()` hook but I've encountered some issues with that for my use case. 

This is purely for navigating to the next and previous article on a detail page, so slightly different to the paginati..

This is why I initially used the results from `useInstantSearch()` as I could get the current index and then the article "either side" of that.

I tried limiting the pagination to 1 article but then I'm not sure I can get the ones "either side"

In this case, the pagination is used purely for fetching the article and then I manually navigate to the details slug.
